qa/test_framework/util.py

```diff
@@ -439,3 +440,46 @@ def get_coinbase_address(node, expected_utxos=None):
     addrs = [a for a in set(addrs) if addrs.count(a) == expected_utxos]
     assert(len(addrs) > 0)
     return addrs[0]
+
+"""
+Perform SC creation, mine mainchain blocks, create genesis info.
+Parameters:
+ - sidechain_id: id of the sidechain to be created
+ - mainchain_node: the mainchain node
+ - secrets: a JSON array of genesis secrets in the following form: [{"secret":"a string", "publicKey":"a string"}...]
+ - amounts: an array of amounts for each secret.
+ 
+Output: an array of two information:
+ - the genesis info used for start the sidechain node
+ - the height of the mainchain block at which the sidechain has been created (useful for future checks of mainchain block reference inclusion)
+
+"""
+def get_genesis_info(sidechain_id, mainchain_node, secrets, amounts):
```

Comment:
Method does a lot of steps inside with different character. Why not to split it to several methods?
`get_genesis_info` and generating blocks -> tx creation -> ...  is confusing.

Reply:
Because all steps are part of the same feature and having several methods increase the coupling between them with an higher impact if they will be modified in the future. Besides, all steps are not used in other methods